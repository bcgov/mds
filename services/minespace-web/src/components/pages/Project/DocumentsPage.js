import React from "react";
import { Row, Col, Typography } from "antd";
import { withRouter } from "react-router-dom";
import PropTypes, { string } from "prop-types";
import DocumentTable from "@/components/common/DocumentTable";

const propTypes = {
  title: PropTypes.string.isRequired,
  documents: PropTypes.arrayOf(PropTypes.object).isRequired,
  onArchivedDocuments: PropTypes.func.isRequired,
};

export const DocumentsPage = (props) => {
  const { projectGuid } = props.match.params;

  return (
    <Row>
      <Col span={24}>
        <Typography.Title level={3}>{props.title}</Typography.Title>
      </Col>
      <Col span={24}>
        <DocumentTable
          documents={props.documents}
          documentParent={props.title}
          canArchiveDocuments={true}
          onArchivedDocuments={props.onArchivedDocuments}
          showVersionHistory={true}
          projectGuid={projectGuid}
        />
      </Col>
    </Row>
  );
};

DocumentsPage.propTypes = propTypes;

export default withRouter(DocumentsPage);
