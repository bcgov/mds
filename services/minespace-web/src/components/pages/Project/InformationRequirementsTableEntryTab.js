import React from "react";
import { withRouter } from "react-router-dom";
import { Row, Col, Typography, Steps, Button, Empty } from "antd";
import PropTypes from "prop-types";
import * as routes from "@/constants/routes";
import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  irt: CustomPropTypes.informationRequirementsTable.isRequired,
  mrcReviewRequired: PropTypes.bool.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectGuid: PropTypes.string,
    }),
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

export const InformationRequirementsTableEntryTab = (props) => {
  const irtExists = Boolean(props?.irt?.irt_guid);
  const projectGuid = props?.irt?.project_guid || props.match.params?.projectGuid;
  const irtGuid = props?.irt?.irt_guid;
  const irtHasChangesRequested = props?.irt?.status_code === "CHR";

  const renderContent = () => {
    const buttonContent = {
      label: irtExists ? "Resume" : "Start",
      link: irtExists
        ? () =>
            props.history.push({
              pathname: `${routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                projectGuid,
                irtGuid
              )}`,
              state: { current: 2 },
            })
        : () =>
            props.history.push(
              `${routes.ADD_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(projectGuid)}`
            ),
    };
    if (irtHasChangesRequested) {
      buttonContent.link = () =>
        props.history.push({
          pathname: `${routes.ADD_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(projectGuid)}`,
          state: { current: 1 },
        });
    }

    const entryGraphic = (
      <Empty
        image={Empty.PRESENTED_IMAGE_SIMPLE}
        imageStyle={{ transform: "scale(2.0)" }}
        description={false}
      />
    );

    return !irtExists ? (
      <div style={{ textAlign: "center" }}>
        <br />
        {entryGraphic}
        <br />
        <Typography.Paragraph>
          <Typography.Title level={5}>Start new IRT submission</Typography.Title>
          Based on your project description, an Information Requirements Table is{" "}
          {props?.mrcReviewRequired ? <b>required</b> : <b>optional</b>}.
        </Typography.Paragraph>
        <Typography.Paragraph>
          Start the final IRT submission process by clicking the button below.
        </Typography.Paragraph>
        <div>
          <Button type="primary" onClick={buttonContent.link}>
            {buttonContent.label}
          </Button>
        </div>
      </div>
    ) : (
      <div style={{ textAlign: "center" }}>
        <br />
        {entryGraphic}
        <br />
        <Typography.Paragraph>
          <Typography.Title level={5}>Resume IRT submission</Typography.Title>
          <div style={{ width: "60%", margin: "0 auto" }}>
            <Steps size="small" current={2}>
              <Steps.Step title="Download Template" />
              <Steps.Step title="Import File" />
              <Steps.Step title="Review & Submit" />
            </Steps>
          </div>
          <br />
          Based on your project description, an Information Requirements Table is{" "}
          {props?.mrcReviewRequired ? <b>required</b> : <b>optional</b>}.
        </Typography.Paragraph>
        <Typography.Paragraph>
          Resume where you left off by clicking the button below.
        </Typography.Paragraph>
        <div>
          <Button type="primary" onClick={buttonContent.link}>
            {buttonContent.label}
          </Button>
        </div>
      </div>
    );
  };

  return (
    <Row>
      <Col span={24}>
        <Typography.Title level={4}>Information Requirements Table</Typography.Title>
      </Col>
      <Col span={24}>{renderContent()}</Col>
    </Row>
  );
};

InformationRequirementsTableEntryTab.propTypes = propTypes;

export default withRouter(InformationRequirementsTableEntryTab);
