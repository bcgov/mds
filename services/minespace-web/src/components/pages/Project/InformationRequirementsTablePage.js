import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Link, withRouter } from "react-router-dom";
import { Row, Col, Button, Typography, Steps } from "antd";
import { ArrowLeftOutlined, DownloadOutlined, HourglassOutlined } from "@ant-design/icons";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import * as routes from "@/constants/routes";
import { ENVIRONMENT } from "@common/constants/environment";
import * as API from "@common/constants/API";

import { getProject, getRequirements } from "@common/selectors/projectSelectors";
import { clearInformationRequirementsTable } from "@common/actions/projectActions";
import {
  fetchProjectById,
  fetchRequirements,
  updateInformationRequirementsTable,
} from "@common/actionCreators/projectActionCreator";
import InformationRequirementsTableCallout from "@/components/Forms/projects/informationRequirementsTable/InformationRequirementsTableCallout";
import { getInformationRequirementsTableDocumentTypesHash } from "@common/selectors/staticContentSelectors";
import IRTDownloadTemplate from "@/components/Forms/projects/informationRequirementsTable/IRTDownloadTemplate";
import IRTFileImport from "@/components/Forms/projects/informationRequirementsTable/IRTFileImport";
import { InformationRequirementsTableForm } from "@/components/Forms/projects/informationRequirementsTable/InformationRequirementsTableForm";

const propTypes = {
  project: CustomPropTypes.project.isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  updateInformationRequirementsTable: PropTypes.func.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  requirements: PropTypes.arrayOf(CustomPropTypes.requirements).isRequired,
  fetchRequirements: PropTypes.func.isRequired,
  clearInformationRequirementsTable: PropTypes.func.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  informationRequirementsTableDocumentTypesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  match: PropTypes.shape({
    params: {
      projectGuid: PropTypes.string,
      tab: PropTypes.string,
    },
  }).isRequired,
  location: PropTypes.shape({
    state: {
      current: PropTypes.number,
    },
  }).isRequired,
};

const tabs = [
  "introduction-and-project-overview",
  "baseline-information",
  "mine-plan",
  "reclamation-closure-plan",
  "modelling-mitigation-discharges",
  "environmental-assessment-predictions",
  "environmental-monitoring",
  "health-safety",
  "management-plan",
];

const StepForms = (
  props,
  state,
  next,
  prev,
  handleTabChange,
  handleIRTUpdate,
  importIsSuccessful,
  downloadIRTTemplate
) => [
  {
    title: "Download Template",
    content: (
      <IRTDownloadTemplate
        downloadIRTTemplate={() =>
          downloadIRTTemplate(
            ENVIRONMENT.apiUrl + API.INFORMATION_REQUIREMENTS_TABLE_TEMPLATE_DOWNLOAD
          )
        }
      />
    ),
    buttons: [
      null,
      <Button id="step1-next" type="primary" onClick={() => next()}>
        Continue to Import
      </Button>,
    ],
  },
  {
    title: "Import File",
    content: (
      <IRTFileImport
        projectGuid={props.project.project_guid}
        informationRequirementsTableDocumentTypesHash={
          props.informationRequirementsTableDocumentTypesHash
        }
        importIsSuccessful={importIsSuccessful}
        downloadIRTTemplate={downloadIRTTemplate}
      />
    ),
    buttons: [
      <>
        {props.project.information_requirements_table?.status_code !== "PRG" ? (
          <>
            <Button
              id="step2-next"
              type="primary"
              onClick={() => {
                props.history.push({
                  pathname: `${routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                    props.project?.project_guid,
                    props.project?.information_requirements_table?.irt_guid
                  )}`,
                  state: { current: 2 },
                });
              }}
              disabled={
                !state.uploadedSuccessfully &&
                !props.project?.information_requirements_table?.irt_guid
              }
            >
              Continue to Review
            </Button>
          </>
        ) : (
          <>
            <Button
              id="step-back"
              type="tertiary"
              className="full-mobile"
              style={{ marginRight: "12px" }}
              onClick={() => prev()}
              disabled={state.submitting}
            >
              Back
            </Button>
            <Button
              id="step2-next"
              type="primary"
              onClick={() => {
                props.history.push({
                  pathname: `${routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                    props.project?.project_guid,
                    props.project?.information_requirements_table?.irt_guid
                  )}`,
                  state: { current: 2 },
                });
              }}
              disabled={
                !state.uploadedSuccessfully &&
                !props.project?.information_requirements_table?.irt_guid
              }
            >
              Continue to Review
            </Button>
          </>
        )}
      </>,
    ],
  },
  {
    title: "Review & Submit",
    content: (
      <>
        {props.project?.information_requirements_table?.status_code === "PRG" ? (
          <Typography.Title level={4}>Review IRT before submission</Typography.Title>
        ) : null}
        <InformationRequirementsTableForm
          project={props.project}
          informationRequirementsTable={props.project?.information_requirements_table}
          requirements={props.requirements}
          tab={props.match?.params?.tab}
          sideMenuOptions={tabs}
          isEditMode={state.isEditMode}
          handleTabChange={handleTabChange}
        />
      </>
    ),
    buttons: [
      <>
        {props.project.information_requirements_table?.status_code === "PRG" ? (
          <>
            <Button
              id="step-back"
              type="tertiary"
              className="full-mobile"
              style={{ marginRight: "24px" }}
              onClick={() => {
                props.history.push({
                  pathname: `${routes.ADD_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                    props.project.project_guid
                  )}`,
                  state: { current: 1 },
                });
              }}
              disabled={props.project?.information_requirements_table?.status_code === "APV"}
            >
              Back
            </Button>
            ,
            <Link
              to={routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                props.project?.project_guid,
                props.project?.information_requirements_table?.irt_guid
              )}
            >
              <Button
                id="submit_irt"
                type="primary"
                htmlType="submit"
                onClick={() => {
                  handleIRTUpdate({ status_code: "REC" }, "IRT submitted ");
                }}
                disabled={state.submitting}
              >
                Submit IRT
              </Button>
            </Link>
          </>
        ) : (
          <>
            <Button
              type="secondary"
              htmlType="submit"
              style={{ marginRight: "24px" }}
              onClick={() => {
                props.history.push({
                  pathname: `${routes.RESUBMIT_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                    props.project?.project_guid,
                    props.project?.information_requirements_table?.irt_guid
                  )}`,
                  state: { current: 1 },
                });
              }}
              disabled={state.submitting}
            >
              Resubmit IRT
            </Button>

            <Button
              type="ghost"
              style={{ border: "none", marginRight: "12px" }}
              className="full-mobile"
              onClick={() =>
                downloadIRTTemplate(
                  ENVIRONMENT.apiUrl + API.INFORMATION_REQUIREMENTS_TABLE_TEMPLATE_DOWNLOAD
                )
              }
            >
              <DownloadOutlined />
              Download IRT template
            </Button>

            <Button type="ghost" style={{ border: "none" }} className="full-mobile">
              <HourglassOutlined />
              File History
            </Button>
          </>
        )}
      </>,
    ],
  },
];

export class InformationRequirementsTablePage extends Component {
  state = {
    current: 0,
    submitting: false,
    isLoaded: false,
    isEditMode: false,
    activeTab: tabs[0],
    informationRequirementsTable: [],
    uploadedSuccessfully: false,
  };

  componentDidMount() {
    this.handleFetchData()
      .then(() => {
        this.setState((prevState) => ({
          current: this.props.location?.state?.current || prevState.current,
        }));
      })
      .then(() => {
        if (this.props.project.information_requirements_table?.status_code === "PRG") {
          this.setState((prevState) => ({
            isEditMode: !prevState.isEditMode,
          }));
        }
      });
  }

  componentWillUnmount() {
    this.props.clearInformationRequirementsTable();
  }

  handleTabChange = (activeTab) => {
    const { projectGuid, irtGuid } = this.props.match.params;
    this.props.history.push({
      pathname: `${routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
        projectGuid,
        irtGuid,
        activeTab
      )}`,
      state: { current: 2 },
    });
  };

  next = () => this.setState((prevState) => ({ current: prevState.current + 1 }));

  prev = () => this.setState((prevState) => ({ current: prevState.current - 1 }));

  importIsSuccessful = () => {
    this.setState((state) => ({
      uploadedSuccessfully: !state.uploadedSuccessfully,
      isEditMode: !state.isEditMode,
    }));
    this.handleFetchData();
  };

  handleFetchData = () => {
    const { projectGuid } = this.props.match?.params;

    return this.props
      .fetchProjectById(projectGuid)
      .then(() => this.props.fetchRequirements())
      .then(() => this.setState({ isLoaded: true }));
  };

  handleIRTUpdate = (values, message) => {
    const projectGuid = this.props.project.project_guid;
    const informationRequirementsTableGuid = this.props.project.information_requirements_table
      .irt_guid;
    this.setState({ submitting: true });
    return this.props
      .updateInformationRequirementsTable(
        {
          projectGuid,
          informationRequirementsTableGuid,
        },
        values,
        message
      )
      .then(() => {
        this.handleFetchData();
        this.setState({ submitting: false });
      });
  };

  downloadIRTTemplate = (url) => {
    const anchor = document.createElement("a");
    anchor.href = url;
    anchor.style.display = "none";
    document.body.append(anchor);
    anchor.click();
    anchor.remove();
  };

  render() {
    const title =
      this.props.project.information_requirements_table?.status_code !== "PRG"
        ? this.props.project?.project_title
        : `Major Mine Submission - ${this.props.project?.project_title}`;

    const Forms = StepForms(
      this.props,
      this.state,
      this.next,
      this.prev,
      this.handleTabChange,
      this.handleIRTUpdate,
      this.importIsSuccessful,
      this.downloadIRTTemplate
    );

    return (
      this.state.isLoaded && (
        <>
          {!this.state.projectActiveTab && (
            <>
              <Row>
                <Col span={24}>
                  <Typography.Title>{title}</Typography.Title>
                </Col>
              </Row>
              <Row>
                <Col span={24}>
                  <Link
                    to={routes.EDIT_PROJECT.dynamicRoute(
                      this.props.project.project_summary.project_guid
                    )}
                  >
                    <ArrowLeftOutlined className="padding-sm--right" />
                    Back to: {this.props.project.project_title} Project Overview page
                  </Link>
                </Col>
              </Row>
              <br />
            </>
          )}
          <Row>
            <Col span={12}>
              <Typography.Title level={2}>Information Requirements Table</Typography.Title>
            </Col>
            <Col span={12}>
              <div style={{ display: "inline", float: "right" }}>
                <p>{Forms[this.props.location.state?.current || this.state.current].buttons}</p>
              </div>
            </Col>
          </Row>
          <Row>
            <Steps current={this.props.location.state?.current || this.state.current}>
              {Forms.map((step) => (
                <Steps.Step key={step.title} title={step.title} />
              ))}
            </Steps>
            <br />
            <br />

            <div>
              <InformationRequirementsTableCallout
                informationRequirementsTableStatus={
                  this.props.project?.information_requirements_table?.status_code || "PRG"
                }
              />
            </div>

            <Col span={24}>
              <div>{Forms[this.props.location.state?.current || this.state.current].content}</div>
            </Col>
          </Row>
        </>
      )
    );
  }
}

const mapStateToProps = (state) => ({
  project: getProject(state),
  requirements: getRequirements(state),
  informationRequirementsTableDocumentTypesHash: getInformationRequirementsTableDocumentTypesHash(
    state
  ),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      clearInformationRequirementsTable,
      fetchProjectById,
      fetchRequirements,
      updateInformationRequirementsTable,
    },
    dispatch
  );

InformationRequirementsTablePage.propTypes = propTypes;

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(InformationRequirementsTablePage)
);
