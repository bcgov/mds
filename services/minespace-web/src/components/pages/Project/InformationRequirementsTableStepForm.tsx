import React from "react";
import { Link, useHistory } from "react-router-dom";
import IRTDownloadTemplate from "@/components/Forms/projects/informationRequirementsTable/IRTDownloadTemplate";
import { ENVIRONMENT, IProject, IRequirement } from "@mds/common";
import * as API from "@mds/common/constants/API";
import { Button, Popconfirm, Typography } from "antd";
import * as routes from "@/constants/routes";
import DownloadOutlined from "@ant-design/icons/DownloadOutlined";
import HourglassOutlined from "@ant-design/icons/HourglassOutlined";
import { IRTFileImport } from "@/components/Forms/projects/informationRequirementsTable/IRTFileImport";
import InformationRequirementsTableForm from "@/components/Forms/projects/informationRequirementsTable/InformationRequirementsTableForm";

interface StepFormsProps {
  submitting: boolean;
  next: () => void;
  prev: () => void;
  handleTabChange: (tab: string) => void;
  handleIRTUpdate: (values: any, message: string) => void;
  handleIRTStatusUpdate: (values: any, message: string) => void;
  importIsSuccessful: (success: boolean, err: string) => void;
  downloadIRTTemplate: (url: string) => void;
  openViewFileHistoryModal: () => void;
  project: IProject;
  requirements: IRequirement[];
  informationRequirementsTableDocumentTypesHash: any;
  uploadedSuccessfully: boolean;
  tabs;
  activeTab: string;
}

const StepForms = ({
  submitting,
  next,
  prev,
  handleTabChange,
  handleIRTUpdate,
  handleIRTStatusUpdate,
  importIsSuccessful,
  downloadIRTTemplate,
  openViewFileHistoryModal,
  project,
  requirements,
  informationRequirementsTableDocumentTypesHash,
  uploadedSuccessfully,
  tabs,
  activeTab,
}: StepFormsProps) => {
  const history = useHistory();

  return [
    {
      key: "downloadIRTTemplate",
      title: "Download Template",
      content: (
        <IRTDownloadTemplate
          downloadIRTTemplate={() =>
            downloadIRTTemplate(
              ENVIRONMENT.apiUrl + API.INFORMATION_REQUIREMENTS_TABLE_TEMPLATE_DOWNLOAD
            )
          }
        />
      ),
      buttons: [
        null,
        <Button key="continue" id="step1-next" type="primary" onClick={() => next()}>
          Continue to Import
        </Button>,
      ],
    },
    {
      key: "importFile",
      title: "Import File",
      content: (
        <IRTFileImport
          projectGuid={project.project_guid}
          informationRequirementsTableDocumentTypesHash={
            informationRequirementsTableDocumentTypesHash
          }
          importIsSuccessful={importIsSuccessful}
          downloadIRTTemplate={downloadIRTTemplate}
        />
      ),
      buttons: [
        <div key="import-buttons">
          <Button
            id="step-back"
            className="full-mobile"
            style={{ marginRight: "12px" }}
            onClick={() => prev()}
            disabled={submitting}
          >
            Back
          </Button>
          <Button
            id="step2-next"
            type="primary"
            onClick={() => {
              history.push({
                pathname: `${routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                  project?.project_guid,
                  project?.information_requirements_table?.irt_guid
                )}`,
                state: { current: 2 },
              });
            }}
            disabled={!uploadedSuccessfully && !project?.information_requirements_table?.irt_guid}
          >
            Continue to Review
          </Button>
        </div>,
      ],
    },
    {
      key: "reviewAndSubmit",
      title: "Review & Submit",
      content: (
        <>
          {project?.information_requirements_table?.status_code === "DFT" ? (
            <>
              <Typography.Title level={4}>Review IRT before submission</Typography.Title>
              <Typography.Paragraph>
                Review imported data before submission. Check the requirements and comments fields
                that are required for the project.
              </Typography.Paragraph>
            </>
          ) : null}

          <InformationRequirementsTableForm
            informationRequirementsTable={project?.information_requirements_table}
            requirements={requirements}
            tab={activeTab}
            sideMenuOptions={tabs}
            handleTabChange={handleTabChange}
          />
        </>
      ),
      buttons: [
        <div key="review-buttons">
          {project.information_requirements_table?.status_code === "DFT" ? (
            <>
              <Button
                id="step-back"
                className="full-mobile"
                style={{ marginRight: "24px" }}
                onClick={() => {
                  history.push({
                    pathname: `${routes.ADD_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                      project.project_guid
                    )}`,
                    state: { current: 1 },
                  });
                }}
              >
                Back
              </Button>
              <Link
                to={routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                  project?.project_guid,
                  project?.information_requirements_table?.irt_guid
                )}
              >
                <Popconfirm
                  placement="topRight"
                  title="Are you sure you want to submit your final IRT, no changes could be made after submitting?"
                  onConfirm={() =>
                    handleIRTStatusUpdate(
                      {
                        status_code: "SUB",
                      },
                      "Successfully submitted final IRT."
                    )
                  }
                  okText="Yes"
                  cancelText="No"
                >
                  <Button id="submit_irt" type="primary">
                    Submit IRT
                  </Button>
                </Popconfirm>
              </Link>
            </>
          ) : (
            <>
              {project.information_requirements_table?.status_code !== "APV" &&
                project.information_requirements_table?.status_code !== "UNR" && (
                  <Button
                    htmlType="submit"
                    style={{ marginRight: "24px" }}
                    onClick={() => {
                      history.push({
                        pathname: `${routes.RESUBMIT_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                          project?.project_guid,
                          project?.information_requirements_table?.irt_guid
                        )}`,
                        state: { current: 1 },
                      });
                    }}
                    disabled={submitting}
                  >
                    Resubmit IRT
                  </Button>
                )}
              <Button
                type="ghost"
                style={{ border: "none", marginRight: "12px" }}
                className="full-mobile"
                onClick={() =>
                  downloadIRTTemplate(
                    ENVIRONMENT.apiUrl + API.INFORMATION_REQUIREMENTS_TABLE_TEMPLATE_DOWNLOAD
                  )
                }
              >
                <DownloadOutlined />
                Download IRT template
              </Button>

              <Button
                type="ghost"
                style={{ border: "none" }}
                className="full-mobile"
                onClick={openViewFileHistoryModal}
              >
                <HourglassOutlined />
                File History
              </Button>
            </>
          )}
        </div>,
      ],
    },
  ];
};

export default StepForms;
