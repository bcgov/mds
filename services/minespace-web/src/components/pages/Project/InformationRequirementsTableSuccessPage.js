import React from "react";
import { withRouter, Link } from "react-router-dom";
import { Row, Col, Typography, Button, Divider } from "antd";
import CheckCircleOutlined from "@ant-design/icons/CheckCircleOutlined";
import ArrowLeftOutlined from "@ant-design/icons/ArrowLeftOutlined";
import PropTypes from "prop-types";
import * as routes from "@/constants/routes";
import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  location: PropTypes.shape({
    state: {
      project: CustomPropTypes.project,
    },
  }).isRequired,
};

export const InformationRequirementsTableSuccessPage = (props) => {
  const renderContent = () => {
    const project = props.location.state?.project || {};
    const {
      project_guid,
      information_requirements_table: { irt_guid },
    } = project;

    return (
      <div style={{ textAlign: "center" }}>
        <>
          <Row>
            <Col span={24}>
              <CheckCircleOutlined className="success-page" />
            </Col>
          </Row>
          <br />
          <Typography.Paragraph>
            <Typography.Title level={5}>Thank you, your IRT has been submitted!</Typography.Title>
            Your submission will soon be reviewed by the ministry.
          </Typography.Paragraph>
          <Typography.Paragraph>
            You will be contacted when the status updates on your submission.
          </Typography.Paragraph>
          <div>
            <p>
              <Link to={routes.EDIT_PROJECT.dynamicRoute(project_guid)}>
                <Button type="primary">Back to Project Overview</Button>
              </Link>
            </p>
            <p>
              <Link
                to={{
                  pathname: routes.REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
                    project_guid,
                    irt_guid
                  ),
                  state: { current: 2 },
                }}
              >
                <Button>View Application</Button>
              </Link>
            </p>
          </div>
        </>
      </div>
    );
  };

  return (
    <>
      <Row>
        <Col span={24}>
          <Typography.Title>{props.location.state?.project.project_title}</Typography.Title>
        </Col>
      </Row>
      <Row>
        <Col span={24}>
          <Link to={routes.EDIT_PROJECT.dynamicRoute(props.location.state?.project.project_guid)}>
            <ArrowLeftOutlined className="padding-sm--right" />
            Back to Project Overview
          </Link>
        </Col>
      </Row>
      <Divider />
      <Row>
        <Col span={24}>
          <Typography.Title level={4}>Information Requirements Table</Typography.Title>
        </Col>
        <Col span={24}>{renderContent()}</Col>
      </Row>
    </>
  );
};

InformationRequirementsTableSuccessPage.propTypes = propTypes;

export default withRouter(InformationRequirementsTableSuccessPage);
