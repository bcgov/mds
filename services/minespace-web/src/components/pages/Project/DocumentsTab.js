import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { withRouter } from "react-router-dom";
import { Row, Col, Tabs, Typography } from "antd";
import PropTypes from "prop-types";
import { formatUrlToUpperCaseString } from "@common/utils/helpers";
import { getProject } from "@common/selectors/projectSelectors";
import { fetchProjectById } from "@common/actionCreators/projectActionCreator";
import customPropTypes from "@/customPropTypes";
import DocumentsPage from "./DocumentsPage";
import { getMineDocuments } from "@common/selectors/mineSelectors";
import ArchivedDocumentsSection from "@common/components/documents/ArchivedDocumentsSection";
import { documentNameColumn, uploadDateColumn } from "@/components/common/DocumentColumns";
import { Feature, isFeatureEnabled } from "@mds/common";
import { MineDocument } from "@common/models/documents/document";
import { renderCategoryColumn } from "@/components/common/CoreTableCommonColumns";
import * as Strings from "@common/constants/strings";

const propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectGuid: PropTypes.string,
    }),
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  project: customPropTypes.project.isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  refreshData: PropTypes.func.isRequired,
  mineDocuments: PropTypes.arrayOf(customPropTypes.mineDocument),
};

const tabs = [
  "project-description",
  "information-requirements-table",
  "major-mine-application",
  isFeatureEnabled(Feature.MAJOR_PROJECT_ARCHIVE_FILE) && "archived-documents",
].filter(Boolean);

export class DocumentsTab extends Component {
  state = {
    activeTab: tabs[0],
  };

  allDocuments = [];

  componentDidMount() {
    this.handleFetchData();
    this.allDocuments.push(this.props.project?.project_summary?.documents);
    this.allDocuments.push(this.props.project?.information_requirements_table?.documents);
    this.allDocuments.push(this.props.project?.major_mine_application?.documents);
  }

  handleFetchData = () => {
    const { projectGuid } = this.props.match?.params;
    return this.props.fetchProjectById(projectGuid);
  };

  handleTabChange = (activeTab) => {
    const { projectGuid } = this.props.match.params;
    const url = `/projects/${projectGuid}/documents/#${activeTab}`;
    this.props.history.push({ pathname: url });

    const a = document.createElement("a");
    a.href = url;
    a.click();
    a.remove();
  };

  render() {
    const documentColumns = [documentNameColumn(), uploadDateColumn()];
    const renderAllDocuments = (docs) => (
      <Row>
        <Col span={24}>
          <div id="project-description">
            <DocumentsPage
              onArchivedDocuments={this.props.refreshData}
              title={formatUrlToUpperCaseString(tabs[0])}
              documents={docs[0]}
            />
          </div>
          <div id="information-requirements-table">
            <DocumentsPage
              onArchivedDocuments={this.props.refreshData}
              title={formatUrlToUpperCaseString(tabs[1])}
              documents={docs[1]}
            />
          </div>
          <div id="major-mine-application">
            <DocumentsPage
              onArchivedDocuments={this.props.refreshData}
              title={formatUrlToUpperCaseString(tabs[2])}
              documents={docs[2]}
            />
          </div>

          <ArchivedDocumentsSection
            titleLevel={3}
            additionalColumns={[
              renderCategoryColumn(
                "category_code",
                "Category",
                Strings.CATEGORY_CODE,
                true
              ),
            ]}
            documentColumns={documentColumns}
            documents={this.props.mineDocuments && this.props.mineDocuments.length > 0
              ? this.props.mineDocuments.map((doc) => new MineDocument(doc)) : []}
          />
        </Col>
      </Row>
    );

    return (
      <Row gutter={16}>
        <Col span={24}>
          <Typography.Title level={2}>All Project Documents</Typography.Title>
        </Col>
        <Col span={24}>
          <Tabs
            tabPosition="left"
            activeKey={this.state.activeTab}
            defaultActiveKey={tabs[0]}
            onChange={(tab) => this.handleTabChange(tab)}
            className="vertical-tabs"
          >
            {tabs.map((tab) => {
              return (
                <Tabs.TabPane
                  tab={formatUrlToUpperCaseString(tab)}
                  key={tab}
                  className="vertical-tabs--tabpane"
                >
                  {renderAllDocuments(this.allDocuments)}
                </Tabs.TabPane>
              );
            })}
          </Tabs>
        </Col>
      </Row>
    );
  }
}

const mapStateToProps = (state) => ({
  project: getProject(state),
  mineDocuments: getMineDocuments(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchProjectById,
    },
    dispatch
  );

DocumentsTab.propTypes = propTypes;

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(DocumentsTab));
