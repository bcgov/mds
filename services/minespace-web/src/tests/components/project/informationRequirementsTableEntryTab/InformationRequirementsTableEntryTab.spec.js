import React from "react";
import { shallow } from "enzyme";
import { InformationRequirementsTableEntryTab } from "@/components/pages/Project/InformationRequirementsTableEntryTab";
import * as MOCK from "@/tests/mocks/dataMocks";

const props = {};

const setupProps = () => {
  props.irt = MOCK.INFORMATION_REQUIREMENTS_TABLE;
  props.match = { params: { projectGuid: "1234-5678-x" } };
  props.history = { push: jest.fn() };
  props.mrcReviewRequired = false;
};

beforeEach(() => {
  setupProps();
});

describe("InformationRequirementsTableEntryTab", () => {
  it("renders properly", () => {
    const component = shallow(<InformationRequirementsTableEntryTab {...props} />);
    expect(component).toMatchSnapshot();
  });
});
