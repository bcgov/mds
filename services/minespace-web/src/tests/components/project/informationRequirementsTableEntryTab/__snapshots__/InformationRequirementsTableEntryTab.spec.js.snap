// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationRequirementsTableEntryTab renders properly 1`] = `
<Row>
  <Col
    span={24}
  >
    <Title
      level={4}
    >
      Information Requirements Table
    </Title>
  </Col>
  <Col
    span={24}
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <Empty
        description={false}
        image={<Simple />}
        imageStyle={
          Object {
            "transform": "scale(2.0)",
          }
        }
      />
      <br />
      <Paragraph>
        <Title
          level={5}
        >
          Resume IRT submission
        </Title>
        <div
          style={
            Object {
              "margin": "0 auto",
              "width": "60%",
            }
          }
        >
          <Steps
            current={2}
            size="small"
          >
            <Step
              title="Download Template"
            />
            <Step
              title="Import File"
            />
            <Step
              title="Review & Submit"
            />
          </Steps>
        </div>
        <br />
        Based on your project description, an Information Requirements Table is
         
        <b>
          optional
        </b>
        .
      </Paragraph>
      <Paragraph>
        Resume where you left off by clicking the button below.
      </Paragraph>
      <div>
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Resume
        </Button>
      </div>
    </div>
  </Col>
</Row>
`;
